client/my-sites/post-relative-time-status/index.jsx

```diff
@@ -21,118 +20,143 @@ class PostRelativeTime extends React.PureComponent {
 	static propTypes = {
 		showPublishedStatus: PropTypes.bool.isRequired,
 		post: PropTypes.object.isRequired,
-		includeNonDraftStatuses: PropTypes.bool,
 		link: PropTypes.string,
 		target: PropTypes.string,
 		gridiconSize: PropTypes.number,
 	};
 
 	static defaultProps = {
-		includeNonDraftStatuses: false,
 		link: null,
 		target: null,
 	};
 
+	/**
+	 * Get the date to be displayed depending on the status of the post
+	 */
 	getTimestamp() {
 		switch ( this.props.post.status ) {
 			case 'new':
 				return null;
+
 			case 'draft':
-			case 'future':
+			case 'trash':
 			case 'pending':
 				return this.props.post.modified;
+
+			case 'future':
 			default:
 				return this.props.post.date;
 		}
 	}
 
-	getDisplayedTimeFromPost( post ) {
+	getDisplayedTimeForLabel() {
 		const moment = this.props.moment;
-
 		const now = moment();
+		const scheduledDate = moment( this.getTimestamp() );
 
-		if ( ! post ) {
-			// Placeholder text: "a few seconds ago" in English locale
-			return now.fromNow();
+		const isScheduledPost = this.props.post.status === 'future';
+		if ( ! isScheduledPost && Math.abs( now.diff( this.getTimestamp(), 'days' ) ) < 7 ) {
+			const time = moment( this.getTimestamp() );
+			return time.fromNow();
 		}
 
-		const { status, modified, date } = post;
-		const time = moment( includes( [ 'draft', 'pending', 'future' ], status ) ? modified : date );
-		if ( now.diff( time, 'days' ) >= 7 ) {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
-		}
-
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		const scheduledTime = scheduledDate.calendar( null, {
+			nextDay: this.props.translate( '[tomorrow at] LT', {
+				comment: 'LT refers to time (eg. 18:00)',
+			} ),
+			sameElse: this.props.translate( '[on] ll [at] LT', {
```

Comment:
I'm not a native English speaker, so I could be wrong, but it's "on" the right preposition to use after "scheduled"? Seems that can denote [when the post was scheduled rather than the actual schedule date](https://english.stackexchange.com/questions/72122/scheduled-on-vs-scheduled-for).